### ISSUE TITLE
API changes concomitant with hardware details redesign

### ISSUE THREAD

**Issue by codersquid** [canonical-web-and-design/certification.ubuntu.com#53]
Currently the app calls /api/v1/certifiedmodeldetails along with /api/v1/certifiedmodeldevices/ to populate the hardware pages.

I would like to change the API calls needed to populate the these pages.

Either of

1) Change the app to stop using the device info in certifiedmodeldetails. The API will stop populating that information because it duplicates the work that the other call does. The app can call certifiedmodeldevices with the category__in={list} if the redesign only calls for showing "important" categories (VIDEO, NETWORK, etc.).

2) OR, Change the app to stop making a call to certifiedmodeldevices and that information will be folded into the certifiedmodeldetails as a list of devices. The device list would be restricted to a default set of device categories, but categories can be overridden with a  'category' query param.

Grouping of devices by category would be done client side rather than server side in both cases. Unless you make the case otherwise. it's 6 of 12 or half a dozen.

TLDR

When creating the certifiedmodeldetails endpoint, my thought was to provide information that goes in the hardware summary so that a separate call to certifiedmodeldevices wouldn't be needed. I was expecting us to get rid of the details section in the near future with the redesign and have a click-through page for all the other categories that aren't salient enough for the hardware summary.. We've had a lot of discussions in the past about dropping the details section in part or in whole.

**Comment by nottrobin** [comment]
This sounds good.... would you be able to try to show what you're thinking of in a PR?
